WordPress/Classes/ViewRelated/Me/App Settings/EncryptedLogTableViewController.swift

```diff
@@ -0,0 +1,114 @@
+
+        let name = WPCrashLoggingProvider.QueuedLogsDidChangeNotification
+        NotificationCenter.default.addObserver(forName: name, object: nil, queue: .main) { _ in
+            self.updateData()
+        }
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        let spacer = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        self.setToolbarItems([spacer, self.toolbarLabel, spacer], animated: animated)
+        tableView.tableFooterView = UIView(frame: .zero) /// hide lines for empty cells
+
+        navigationController?.setToolbarHidden(false, animated: animated)
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        self.setToolbarItems(nil, animated: animated)
+        navigationController?.setToolbarHidden(true, animated: animated)
+    }
+
+    // MARK: UITableViewController Data Source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return logs.count
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        let log = logs[indexPath.item]
+        let cell = tableView.dequeueReusableCell(withIdentifier: "reuseIdentifier", for: indexPath)
+        cell.textLabel?.text = log.uuid
+
+        if let date = try? FileManager.default.attributesOfItem(atPath: log.url.path)[.creationDate] as? Date {
+            cell.detailTextLabel?.text = DateFormatter.localizedString(from: date, dateStyle: .short, timeStyle: .short)
+        }
+        else {
+            cell.detailTextLabel?.text = "Unknown"
+        }
+
+        return cell
+    }
+
+    // MARK: Internal Helpers
+    private func updateData() {
+        self.logs = CrashLogging.eventLogging?.queuedLogFiles ?? []
+        self.tableView.reloadData()
+
+        if let date = CrashLogging.eventLogging?.uploadsPausedUntil {
+            let dateString = DateFormatter.localizedString(from: date, dateStyle: .short, timeStyle: .medium)
+            self.toolbarLabel.title = "Paused until \(dateString)"
+        }
+        else {
+            self.toolbarLabel.title = self.logs.isEmpty ? "All Logs Uploaded" : "Running"
+        }
+    }
+
+    @objc
+    private func addEncryptedLog() {
+
+        do {
+            /// For now, just enqueue any file – doesn't have to be the log
+            let data = try Data(contentsOf: Bundle.main.url(forResource: "acknowledgements", withExtension: "html")!)
+
+            let url = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+            try data.write(to: url)
+
+            try CrashLogging.eventLogging?.enqueueLogForUpload(log: LogFile(url: url))
+            self.updateData()
```

Comment:
Is this `updateData` call necessary? Wouldn't the `QueuedLogsDidChangeNotification` observer catch that?

Typically, I wouldn't expect the internals of `enqueueLogForUpload` to happen in the main thread in which case updating the data immediately after in the main thread would not do anything. This might not be the case here, but I think it's good to structure the code around that. (we could even move the whole log creation to a bg thread, but that might not be super important for a debug page)

Reply:
>Is this updateData call necessary? Wouldn't the QueuedLogsDidChangeNotification observer catch that?

True – removed!

> Typically, I wouldn't expect the internals of enqueueLogForUpload to happen in the main thread in which case updating the data immediately after in the main thread would not do anything. This might not be the case here, but I think it's good to structure the code around that. (we could even move the whole log creation to a bg thread, but that might not be super important for a debug page)

I think the right place to do this work would be to move[ the body of this method](https://github.com/Automattic/Automattic-Tracks-iOS/blob/feature/encrypted-logging-master/Automattic-Tracks-iOS/Event%20Logging/EventLogging.swift#L10) into a background thread (which is fairly easy). I'm just not sure we're saving much – APFS is copy-on-write, so copies are really cheap to make. If it were ever a problem though, it'd be an easy change. WDYT?

Reply:
Sounds good - I trust your judgement :)